sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt

```diff
@@ -318,10 +332,22 @@ internal object KlaviyoApiClient : ApiClient {
          * Re-queue the job to run again after [flushInterval] milliseconds
          */
         private fun requeue() {
-            Registry.log.verbose("Retrying network batch in $flushInterval")
+            Registry.log.verbose("Network batch will run in $flushInterval ms")
             force = false
+            enqueuedTime = Registry.clock.currentTimeMillis()
```

Comment:
Whats the thought behind now updating the enqueued time whenever an item is re-queued, instead of having it represent when the request was initially made?

Reply:
Yeah good catch. Its kind of a technicality, but this change makes the tests better match actual behavior on the device.

Previously, I was using `flushInterval *= request.attempts + 1`, the idea being that the flush interval is incrementing relative to the initial queue time like so:
- First attempt is 10s after queueing
- Second attempt is 20s after queueing, but 10s after the previous attempt
- Third attempt is 60s after queueing, but 30s after the prior attempt
- etc

And thats what my test was simulating. But that didn't match the true behavior on the device, because below I'm doing `postDelayed(flushInterval)`, which will only post the job after that _total_ wait time. So in reality the behavior was
- First attempt is 10s after queueing
- Second attempt is 30s after queueing, 20s after the previous attempt
- Third attempt is 90s after queueing, 60s after the prior attempt
- etc

Reply:
And actually, re-visiting this is making me realize I think I could leave the initial queued time immutable as before, but use `+= computeRetryInterval(request.attempts)` instead

Reply:
Either way I dont think I really care if `enqueuedTime` is immutable or not. I just thought it seemed a little misleading but we dont have any use for an original queue time being stored on a request right now and I dont think there's any particular reason we should care.

Maybe there could be something like dropping certain really old requests given the original queue time (user returns to app after months of not using it triggering some redundant data to send over)

Reply:
Yeah I think I see your point. In this case, `enqueuedTime` just refers to the batch job running on the background thread. It could be valuable to track enqueued time on the ApiRequest object (and serialize that to the persistent store JSON) though. 
Using `+=` and leaving `enqueuedTime` immutable doesn't actually work. I think I'd opt to keep as is until perhaps we add parallel network request and/or update to use co-routines or other async method.